python.md

```diff
@@ -957,6 +958,23 @@ For example, a test should not assert that there are only a certain number of mo
 instances in the database, as a transactional test (which does commit to the same DB) 
 running concurrently may have created some.
 
+### <a name="test-isolation">Freeze time for tests</a>
+
+Don't let tests or the system-under-test call the system clock unless it is
+being explicitly controlled using a tool like [freezegun](https://github.com/spulec/freezegun).
+
+This guards against a whole class of date-related test bugs which often manifest
+themselves if your test-suite runs in the hours before or after mignight. Often
+these are caused by DST-offsets where a datetime in UTC has a different date to
+one in the local timezone.
+
+For unit tests, it's best to design functions and classes to have date or
+datetimes injected so freezegun isn't necessary.
+
+For integration or functional tests, wrap the fixture creation and test
+invocation in the freezegun wrapper to give tight control of what the system
+clock calls will return.
+
```

Comment:
```suggestion
Don't use this to bypass problematic times in tests and thus avoid testing time sensitive
code around, for example, day and DST changes, rather use it to know what to expect.
Our code needs to cope with the non-linearities of time, which are (probably only
partially) detailed [here](http://www.creativedeletion.com/2015/01/28/falsehoods-programmers-date-time-zones.html)
```

Reply:
Thanks JT - have added a para similar to yours.